### PR TITLE
persist: add microbenchmarks for Postgres-backed Consensus

### PR DESCRIPTION
**Pull request by ruchirK**
### Motivation
```
This commit adds the postgres backed implementation of Consensus to the Consensus
microbenchmarks. Tested on a r5a.4xl machine against a Aurora Postgresql instance
backed by 2 db.r5.8xlarge instances (instance type chosen arbitrarily).

consensus/compare_and_set/postgres/64KiB
                        time:   [6.3041 ms 6.3679 ms 6.4322 ms]
                        thrpt:  [9.7167 MiB/s 9.8148 MiB/s 9.9141 MiB/s]
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild
```

### Testing

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

**Comment by ruchirK** [comment]
The first two commits are #11461

File: src/persist/benches/consensus.rs
Comment (danhhz) [inline src/persist/benches/consensus.rs:18]: i'm intentionally avoiding using the term Runtime in the new persist impl so that people outside the team are less confused by our naming, but it also means we don't have to rename all these anymore

**Review by danhhz — APPROVED** [review]
last commit LGTM modulo one nit

**Comment by ruchirK** [comment]
tftr! merging on green!

**Comment by danhhz** [comment]
don't forget to rebase on top of a merged #11461 before you hit the auto merge button! :D

**Comment by ruchirK** [comment]
> don't forget to rebase on top of a merged https://github.com/MaterializeInc/materialize/pull/11461 before you hit the auto merge button! :D

yup double checked before hitting auto merge :)

**Comment by ruchirK** [comment]
the clippy failures are from some vendored potentially generated .cc files for prost - Nikhil says this should go away on retry so thats what I'm trying first

**Comment by danhhz** [comment]
> > don't forget to rebase on top of a merged #11461 before you hit the auto merge button! :D
> 
> yup double checked before hitting auto merge :)

looks like the two commits from #11461 are still in here?

**Comment by ruchirK** [comment]
oh i see what you mean. I thought github would correctly handle the case if a prefix of commits was already merged (I had just made sure to rebase this branch onto the correct version of 11461). I'll rebase onto main

**Comment by ruchirK** [comment]
alright now we're merging on green!

**Pull request merged**
